lib/cargoClippy.nix

```diff
@@ -19,5 +19,7 @@ mkCargoDerivation (args // {
 
   buildPhaseCargoCommand = "cargoWithProfile clippy ${cargoExtraArgs} ${cargoClippyExtraArgs}";
 
+  doInstallCargoArtifacts = false;
```

Comment:
Why was this change needed?

I don't agree with changing the defaults here because `cargoClippy` is intended to be allow chaining with other builds by default

Reply:
I observed this to be a large slowdown in practice - specifically around copying and patching interpreter paths in the output. My intuition was that folks are using cargoClippy as a linting task, and so like cargoDoc in the last version, it was an oversight which would primarily affect new users.

What would cargoClippy chaining look like for you? Is there an example?

Reply:
While I agree that cargo clippy should be a a leaf node 99% of the time, if someone has a good reason to run it other way it should work. It technically can produce artifacts (same as `cargo check` iirc).

If I read the code here right this change doesn't just change the default, but hardcodes `false`.

Probably best as a separate PR with changelog info about change in behavior.

Reply:
Yep if you'd like to propose a different default value here feel free to submit a separate PR for it (then we can at least have a more focused discussion on it!)